shacl12-core/index.html

```diff
@@ -1587,267 +1585,230 @@ <h3>Property Shapes</h3>
+					This SHACL Core specification does not define any <a>node expression types</a> but defines how they are generally handled.
+					<span class="todo">TODO: Add link to shacl12-node-expr once that is stable.</span>
+				</p>
+				<div class="def">
+					<div class="def-header">TEXTUAL DEFINITION</div>
+					A <dfn>node expression</dfn> is a function with exactly one <dfn>node expression type</dfn> as its type.
+					Each <a>node expression type</a> has an <a>IRI</a>.
+					Node expression types can declares one or more <dfn>node expression parameters</dfn>.
+					Each of these parameters has an <a>IRI</a>.
+					One of these parameters can be the <dfn>key parameter</dfn> that uniquely identifies the node expression type.
+					<br />
+					<br />
+					During <dfn>evaluation</dfn> a node expression takes the following input:
+					<ul>
+						<li>A <a>graph</a>, called the <dfn>active graph</dfn>.</li>
+						<li>An optional <a>node</a> called the <dfn>input focus node</dfn>.</li>
+						<li>One <a>node</a> as value for each of the declared parameters, optional for any but the key parameter unless stated otherwise in the definition of the node expression type.</li>
+					</ul>
+					The result of the evaluation of a node expression is a list of <a>nodes</a> (possibly empty and with duplicates) called the <dfn>output nodes</dfn>.
+					The evaluation may also result in an <dfn>evaluation failure</dfn>.
+				</div>
+				<p class="syntax">
+					<span data-syntax-rule="node-expression-node">Each node expression is represented by one <a>node</a>.</span>
+					<span data-syntax-rule="node-expression-iri">A node expression that is an <a>IRI</a> evaluates to the list consisting of exactly this IRI.</span>
+					<span data-syntax-rule="node-expression-literal">A node expression that is a <a>literal</a> evaluates to the list consisting of exactly this literal.</span>
+					<span data-syntax-rule="node-expression-blank-nodes">All other node expressions are <a>blank nodes</a>.</span>
+					<span data-syntax-rule="node-expression-type">The <a>node expression type</a> of a node expression is determined by
+					the presence of the value of the key parameter at the blank node.</span>
+				</p>
+				<p><em>The remainder of this section is informative.</em></p>
+				<aside class="example" title="A dynamically computed property using a node expression based on a SPARQL query">
+					<p>
+						Here is an example use of a node expression based on <a href="shacl12-sparql">SHACL-SPARQL</a>, computing
+						the values of a property shape for the property "full name" as the concatenation of <code>ex:firstName</code>,
+						a space and the <code>ex:lastName</code>.
+					</p>
+					<div class="shapes-graph">
+						<div class="turtle">
+ex:Person-fullName
+	a sh:PropertyShape ;
+	sh:name "full name" ;
+	sh:path <b>[
+		sh:select """
+			PREFIX ex: &lt;http://example.org/ns#&gt;
+			SELECT ?fullName
```

Comment:
In 1.0 uses of SHACL-SPARQL, the `SELECT` clause binds `$this`.  How should this named variable `?fullName` be bound in SHACL?

Reply:
The engine will just return all bindings for the first variable in the result set. But details of this would be written up in the SHACL-SPARQL spec.